crates/matrix-sdk/src/room/mod.rs

```diff
@@ -1732,12 +1739,14 @@ impl Room {
     #[instrument(skip_all)]
     pub fn send_attachment<'a>(
         &'a self,
-        body: &'a str,
+        filename: &'a str,
         content_type: &'a Mime,
         data: Vec<u8>,
         config: AttachmentConfig,
+        caption: Option<String>,
+        formatted: Option<FormattedBody>,
```

Comment:
Could `caption` go into the `AttachmentConfig`, since it already contains other extra information like the thumbnail URL? Same question for `formatted` (if it's the `formatted_caption` only; if it's used in other contexts, let's keep it separate).